src/modules/emit_sound.rs

```diff
@@ -0,0 +1,219 @@
+//! `bxt_emit_sound`
+
+use std::ffi::CString;
+use std::num::ParseFloatError;
+use std::ptr::NonNull;
+use std::str::FromStr;
+
+use super::Module;
+use crate::handler;
+use crate::hooks::engine;
+use crate::modules::commands::Command;
+use crate::utils::*;
+
+pub struct EmitSound;
+impl Module for EmitSound {
+    fn name(&self) -> &'static str {
+        "bxt_emit_sound"
+    }
+
+    fn description(&self) -> &'static str {
+        "Exposing some sound functions."
+    }
+
+    fn commands(&self) -> &'static [&'static Command] {
+        static COMMANDS: &[&Command] = &[
+            &BXT_EMIT_SOUND_STOP_SOUND_EXCEPT,
+            &BXT_EMIT_SOUND_DYNAMIC,
+            &BXT_EMIT_SOUND,
+        ];
+        COMMANDS
+    }
+
+    fn is_enabled(&self, marker: MainThreadMarker) -> bool {
+        engine::SV_StartSound.is_set(marker)
+            && engine::S_StartDynamicSound.is_set(marker)
+            && engine::S_StopSound.is_set(marker)
+            && engine::S_PrecacheSound.is_set(marker)
+    }
+}
+
+#[derive(Default)]
```

Comment:
If this default is just for FromStr, then better remove it and in FromStr use the following pattern:
```
let volume = ...;
let from = ...;
...

Ok(SoundInfo {
    volume,
    from,
    ...
})
```